e2e/wdio/platform/tests/button.e2e-spec.ts

```diff
@@ -0,0 +1,71 @@
+import { ButtonPo } from '../pages/button.po';
+import {
+    getAttributeByName,
+    getElementArrayLength,
+    getElementTitle,
+    isElementClickable,
+    scrollIntoView,
+} from '../../driver/wdio';
+
+describe('Button test suite:', function() {
+    const buttonPage = new ButtonPo();
+    const {
+        typeButtons, sizeButtons, iconButtons, stateButton, disableStateButtons, truncatedButton  } = buttonPage;
+
+    beforeAll(() => {
+        buttonPage.open();
+    }, 1);
+    
+        it('verify clickable buttons types', () => {
+            const typeButtonsLength = getElementArrayLength(typeButtons);
+            for (let i = 0; i < typeButtonsLength; i++) {
+                scrollIntoView(typeButtons, i);
+                expect(isElementClickable(typeButtons, i)).toBe(true, 'type button with index ${i} not clickable');
+            }
+        });
+
+        it('verify clickable size buttons', () => {
+            const sizeButtonsLength = getElementArrayLength(sizeButtons);
+            for (let i = 0; i < sizeButtonsLength; i++) {
+                scrollIntoView(sizeButtons, i);
+                expect(isElementClickable(sizeButtons, i)).toBe(true, 'size button with index ${i} not clickable');
+            }
+        });
+
+        it('verify buttons with icons', () => {
+            const iconButtonsLength = getElementArrayLength(iconButtons);
+            for (let i = 0; i < iconButtonsLength; i++) {
+                scrollIntoView(iconButtons, i);
+                expect(isElementClickable(iconButtons, i)).toBe(true, 'icon button with index ${i} not clickable');
+            }
+        });
+
+        it('verify state buttons', () => {
+            scrollIntoView(stateButton);
+            expect(isElementClickable(stateButton)).toBe(true, 'state button with index not clickable');
+        });
+
+        it('verify disable state buttons', () => {
+            expect(getAttributeByName(disableStateButtons, 'aria-disabled')).toEqual('true');
+            expect(getAttributeByName(disableStateButtons, 'disabled', 1)).toEqual('true');
```

Comment:
here you get two different attributes for one check, better to get class of both buttons and check that it contains 'is-disabled'
`expect(getElementClass(disableStateButtons)).contain('is-disabled);`
`expect(getElementClass(disableStateButtons, 1)).contain('is-disabled);`

Reply:
+ add error message if assertion fails